YololCompetition/Services/Challenge/DbChallenges.cs

```diff
@@ -167,5 +168,13 @@ public async Task<int> Delete(ulong challengeId)
                 return null;
             return Challenge.Read(reader);
         }
+
+        public async Task<Challenge?> GetChallengesByEndTime(ulong EndUnixTime)
+        {
+            await using var cmd = _database.CreateCommand();
+            cmd.CommandText = "SELECT * FROM Challenges WHERE EndUnixTime >= @EndTime";
+            cmd.parameters.Add(new SqliteParameter("@EndTime", DbType.UInt64) { Value = EndUnixTime });
+            return await cmd.ExecuteNonQueryAsync();
```

Comment:
There are a number of issues here:
 - You should probably use the `SqlAsyncResult` (see `GetPending` for example).
 - The name impliues to be it gets a challenge by an exact end time, instead this will fetch all challenges ending after a given time.
 - They won't be in order!
 - The final call is `Execute**_Non_**QueryAsync`, so I'm not sure what this will do or if it actually compiles.

Reply:
Good shouts. Fixed.